hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -504,6 +505,11 @@ int QuectelNcpClient::getSignalQuality(CellularSignalQuality* qual) {
     return SYSTEM_ERROR_NONE;
 }
 
+int QuectelNcpClient::setRegistrationTimeout(unsigned timeout) {
+    registrationTimeout_ = std::max(timeout, REGISTRATION_CHECK_INTERVAL);
```

Comment:
@monkbroc @eberseth @avtolstoy This `registrationTimeout_` value should be limited to >= 5 minutes.  As written here, the registration timeout could be reduced to 15 seconds.  Simply need to change `REGISTRATION_CHECK_INTERVAL` to `REGISTRATION_TIMEOUT`.  This needs to be changed in both the `quectel_ncp_client.cpp` and `sara_ncp_client.ncp`

Reply:
OK. There was no minimum discussed so we decided to use the 15 second check interval, but I understand that using 5 minutes makes sense if we only allow extending the registration timeout.

Reply:
https://github.com/particle-iot/device-os/pull/2014